launchlib/launcher.go

```diff
@@ -33,41 +33,53 @@ const (
 	ExecPathBlackListRegex = `[^\w.\/_\-]`
 )
 
-type ServiceCommands struct {
-	Primary      *exec.Cmd
-	SubProcesses map[string]*exec.Cmd
+type ServiceCmds struct {
+	Primary             *exec.Cmd
+	PrimaryOutputFile   string
+	SubProcs            map[string]*exec.Cmd
+	SubProcsOutputFiles map[string]string
```

Comment:
I don't think these output file definitions belong here.  One flag is the fact that elsewhere in the code we now have an "impossible edge case" which can only be hit if other code is buggy but has no direct compilation tie to the construction.

We could avoid this by making new `ServiceCmds` structures through a `NewServiceCmds` method, and additionally have the structure itself responsible for starting commands (so we can ensure ahead of time everything is set), but then that explicitly isn't how `go-java-launcher` starts things. 

As such, to me, it seems to make more sense to define this as part of the `go-init` logic.

Reply:
> elsewhere in the code we now have an "impossible edge case" which can only be hit if other code is buggy but has no direct compilation tie to the construction

Can you elaborate here? I'm not sure what you're getting at

Reply:
https://github.com/palantir/go-java-launcher/pull/49/files#diff-9c3962318b8e21e39835af6f2183c281R108